www-misc/buku/buku-4.1.ebuild

```diff
@@ -0,0 +1,103 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5,6} )
+
+inherit bash-completion-r1 distutils-r1
+
+DESCRIPTION="Powerful command-line bookmark manager"
+HOMEPAGE="https://github.com/jarun/Buku"
+
+SRC_URI="https://github.com/jarun/Buku/files/1319933/bookmarks.zip"
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/jarun/Buku"
+else
+	# Tests are not working with the pypi version (https://github.com/jarun/Buku/issues/325)
+	SRC_URI="${SRC_URI} https://github.com/jarun/Buku/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+	# SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz -> ${P}.tar.gz"
+
+	# Github folder name starts with capital B
+	S="${WORKDIR}/${P^}"
+
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="test server"
+
+CLIENT_DEPEND="dev-python/beautifulsoup:4[${PYTHON_USEDEP}]
+	dev-python/certifi[${PYTHON_USEDEP}]
+	dev-python/cryptography[${PYTHON_USEDEP}]
+	dev-python/html5lib[${PYTHON_USEDEP}]
+	>=dev-python/urllib3-1.13.1[${PYTHON_USEDEP}]"
+
+SERVER_DEPEND="dev-python/arrow[${PYTHON_USEDEP}]
+	dev-python/click[${PYTHON_USEDEP}]
+	dev-python/flask[${PYTHON_USEDEP}]
+	dev-python/flask-admin[${PYTHON_USEDEP}]
+	dev-python/flask-api[${PYTHON_USEDEP}]
+	dev-python/flask-bootstrap[${PYTHON_USEDEP}]
+	dev-python/flask-paginate[${PYTHON_USEDEP}]
+	dev-python/flask-wtf[${PYTHON_USEDEP}]
+	dev-python/requests[${PYTHON_USEDEP}]
+	dev-python/werkzeug[${PYTHON_USEDEP}]"
+
+RDEPEND="${CLIENT_DEPEND}
+	server? ( ${SERVER_DEPEND} )"
+
+DEPEND="${RDEPEND}
+	test? (
+		dev-python/attrs[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		dev-python/py[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pyyaml[${PYTHON_USEDEP}]
+		dev-python/vcrpy[${PYTHON_USEDEP}]
+	)
+"
+
+python_prepare_all() {
+	distutils-r1_python_prepare_all
+
+	if ! use server; then
+		sed -i 's/, '\''bukuserver=bukuserver.server:cli'\''\]/\]/g' setup.py || die
```

Comment:
Please use at least `""` to avoid escape hell. Also, any chance to make those seds less fragile?

Reply:
I replaced it with a patch. I think this way it is more obvious what is being changed and easier to maintain.